### PR TITLE
Fix check code format wf

### PR DESCRIPTION
**Pull request by dlesnoff**
The directory `.nim_runtime` created by the new version of the github action `setup-nim-action` must be escaped by all commands including nim prettyfy.

I can not merge this PR myself, so please have a look at it @Panquesito7  as soon as it is validated by e.g. @vil02

### CODE REVIEW

File: .github/workflows/check_code_format.yml [hunk 1]
```diff
@@ -22,14 +22,17 @@ jobs:
...
       - name: Fail if needs reformatting
         run: |
+          echo ".nim_runtime" > .gitignore
           if [[ $(git status --porcelain) ]]; then
              echo "please reformat/prettyfy these files:"
              git status --porcelain=v1
+             rm -vf .gitignore
              exit 1
           fi
+          rm -vf .gitignore
```
Comment (vil02) [inline .github/workflows/check_code_format.yml:37]: Since you add `.nim_runtime` to `.gitignore`, are all of these changes needed?

Comment (dlesnoff) [reply .github/workflows/check_code_format.yml:37]: Sorry but I will avoid the .gitignore file finally, let me push a last change.

File: .github/workflows/check_code_format.yml [hunk 2]
```diff
@@ -22,8 +22,8 @@ jobs:
 
       - name: Format code
         run: |
-          git clean -f -x -d
-          nim prettyfy
+          git clean -f -x -d -e ".nim_runtime"
+          nim prettyfy --excludePath=".nim_runtime"
```
Comment (vil02) [inline .github/workflows/check_code_format.yml:26]: Why not just to run the `setup-nim-action` with [`nim-install-directory`](https://github.com/jiro4989/setup-nim-action/blob/1b7e2433060960eab05e25faa2c9cf1ff69b8cfa/action.yml#L12C3-L12C24) set to some directory _outside_ the workspace?

Comment (dlesnoff) [reply .github/workflows/check_code_format.yml]: Due to an issue in setup-nim-action apparently: https://github.com/jiro4989/setup-nim-action/issues/499

File: .github/workflows/check_code_format.yml [hunk 3]
```diff
@@ -19,6 +19,10 @@ jobs:
           fetch-depth: 0
 
       - uses: jiro4989/setup-nim-action@v2
+        with:
+          nim-version: stable
```
Comment (vil02) [inline .github/workflows/check_code_format.yml:23]: I am not sure if this is needed.
```suggestion
```

**Review by vil02 — APPROVED** [review]
My understanding is that we are lucky: because of:
https://github.com/TheAlgorithms/Nim/blob/e65658c36415509ba4a63c1bee9298d74db040d4/config.nims#L18
other workflows using `setup-nim-action` do not need an update.

File: .github/workflows/check_code_format.yml (hunk 3 above)
Comment (dlesnoff) [reply .github/workflows/check_code_format.yml:23]: Yes it isn't. I just prefer writing it down for future changes.

**Comment by dlesnoff** [comment]
@vil02 Would you like to become a maintainer of this repository? See #9 to apply.
You are already quite involved in the project as a contributor. It's not a job. There are no constraints, no responsibilities, no salary. It's just a matter of write access.
I can't contribute much myself if nobody is going to merge my pull requests.

**Comment by vil02** [comment]
@dlesnoff I wrote you an e-mail.

**Pull request merged**